webapp/javascript/redux/reducers/continuous/state.ts

```diff
@@ -25,33 +25,14 @@ type SingleView =
       annotations: Annotation[];
     };
 
-type TagExplorerView =
-  | {
-      type: 'pristine';
-      groups: Groups;
-      groupByTag: string;
-      groupByTagValue: string;
-    }
-  | {
-      type: 'loading';
-      groups: Groups;
-      groupByTag: string;
-      groupByTagValue: string;
-    }
-  | {
-      type: 'loaded';
-      groups: Groups;
-      groupByTag: string;
-      activeTagProfile?: Profile;
-      groupByTagValue: string;
-    }
-  | {
-      type: 'reloading';
-      groups: Groups;
-      groupByTag: string;
-      activeTagProfile?: Profile;
-      groupByTagValue: string;
-    };
+type TagExplorerView = {
+  groupsLoadingType: 'pristine' | 'loading' | 'loaded' | 'reloading';
+  groups: Groups;
+  groupByTag: string;
+  groupByTagValue: string;
+  activeTagProfile?: Profile;
+  activeTagProfileLoadingType: 'pristine' | 'loading' | 'loaded' | 'reloading';
+};
```

Comment:
I like the original way since it forces you to think about the different states. For example, if type is `pristine`, it means you have no data, so you have to design your UI around it.

Although with the way it was implemented, the types didn't really indicate that, so based on that your refactor makes sense.